### PR TITLE
fix(gatsby): remove redundant dispatch of query extraction error

### PR DESCRIPTION
**Pull request by vladar**
## Description

This PR fixes a bug where we incorrectly mark a component as having a query extraction error. Very edge case, pretty well explained in the test case:

```
This is a regression test for query extraction bug in Gatsby.

All of the following is required to hit this bug:

1. `gatsby-plugin-mdx` should be installed + at least one other plugin implementing `preprocessSource`
2. This plugin must be listed before `gatsby-plugin-mdx` in `gatsby-config.js`
3. MDX file must contain the word `graphql`
```

MDX is not requirement - any two plugins implementing `preprocessSource` could hit this. But it was easier to reproduce and write a test case with mdx.

### CODE REVIEW

File: e2e-tests/mdx/src/pages/regressions/index.mdx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+This is a regression test for query extraction bug in Gatsby.
+
+All of the following is required to hit this bug:
+
+1. There should be at least one other plugin implementing `preprocessSource`
+2. This plugin must be listed before `gatsby-plugin-mdx` in `gatsby-config.js`
+3. MDX file must contain word `graphql`
```
Comment (pieh) [inline e2e-tests/mdx/src/pages/regressions/index.mdx:7]: I assume `graphql` word in this file is important to this test - should maybe add comment saying to not mess with content of this file (or at very least not remove `graphql` word)? Just to add context for any future person that would stumble on this file

**Review by pieh — APPROVED** [review]
First of - "wat"
Secondly - well found ;) LGTM, have one not blocking comment - up to you if you think it needs addressing or not

**Review by pieh — APPROVED** [review]

**Comment by vladar** [comment]
Published in `gatsby@2.29.2`
